src/Grammar.js

```diff
@@ -146,52 +146,70 @@ Grammar.prototype = {
       throw new Error(
           'Why would anyone want to generate a recipe for the ' + this.name + ' grammar?!?!');
     }
-    var sb = new common.StringBuffer();
-    if (optVarName) {
-      sb.append('var ' + optVarName + ' = ');
-    }
-    sb.append('(function() {\n');
 
-    // Include the supergrammar in the recipe if it's not a built-in grammar.
-    var superGrammarDecl = '';
-    if (!this.superGrammar.isBuiltIn()) {
-      sb.append(this.superGrammar.toRecipe('buildSuperGrammar'));
-      superGrammarDecl = '    .withSuperGrammar(buildSuperGrammar.call(this))\n';
-    }
-    sb.append('  var decl = this.newGrammar(' + JSON.stringify(this.name) + ')\n');
+    var recipe = [
+      'grammar',
+      {}, // meta-info
```

Comment:
I'd stick this in a var named `metaInfo`, then you can eliminate the comment, and the code below that uses `recipe[1]` will be a bit more self-documenting.